framework/src/transfers/MultiAppFieldTransfer.C

```diff
@@ -55,6 +63,58 @@ MultiAppFieldTransfer::initialSetup()
     for (auto & from_var : getFromVarNames())
       variableIntegrityCheck(from_var);
   }
+
+  // Convert block names to block IDs, fill with ANY_BLOCK_ID if unspecified
+  if (_has_block_restrictions)
+  {
+    const FEProblemBase * from_problem;
+    const FEProblemBase * to_problem;
+
+    if (_current_direction == FROM_MULTIAPP)
+    {
+      // Subdomain and variable type information is shared on all subapps
+      from_problem = &getFromMultiApp()->appProblemBase(0);
+      to_problem = &getFromMultiApp()->problemBase();
+    }
+    else if (_current_direction == TO_MULTIAPP)
+    {
+      from_problem = &getToMultiApp()->problemBase();
+      to_problem = &getToMultiApp()->appProblemBase(0);
+    }
+    else
+    {
+      from_problem = &getFromMultiApp()->appProblemBase(0);
+      to_problem = &getToMultiApp()->appProblemBase(0);
+    }
+
+    const auto & from_block_names = getParam<std::vector<SubdomainName>>("from_blocks");
+    if (from_block_names.size())
+      _from_blocks = from_problem->mesh().getSubdomainIDs(from_block_names);
+    else
+      _from_blocks = {Moose::ANY_BLOCK_ID};
+
+    const auto & to_block_names = getParam<std::vector<SubdomainName>>("to_blocks");
+    if (to_block_names.size())
+      _to_blocks = to_problem->mesh().getSubdomainIDs(to_block_names);
+    else
+      _to_blocks = {Moose::ANY_BLOCK_ID};
+
+    // Forbid block restriction on nodal variables as currently not supported
+    if (from_block_names.size())
+      for (auto & from_var : getFromVarNames())
+        if (from_problem
+                ->getVariable(
+                    0, from_var, Moose::VarKindType::VAR_ANY, Moose::VarFieldType::VAR_FIELD_ANY)
+                .isNodal())
```

Comment:
Looks like we're defining `isNodal()` via `_continuity == C_ZERO || _continuity == C_ONE`, so this will error properly on the "hybrid" variables we discussed with both Node and Elem dofs ... but is that test an atavism predating `H_CURL` continuity?  It looks like it'll be giving a false negative for `H_CURL` or `SIDE_DISCONTINUOUS` (or `H_DIV`, when we get it) continuity types.